### ISSUE TITLE
[spoilers] day06-bash logic question

### ISSUE THREAD

**Issue by tholok97** [Arxcis/adventofcode2018#39]
During [the solve in Bash for day6](https://github.com/Arxcis/adventofcode2018/blob/master/day06-chronal-coordinates/main.bash) I think made a false assumption about the area to search while looking for answers to part 1. It **happened to work out with the input I had**, so I did not prioritize fixing it at the time. I think I know a way to fix it, but my fix seems a little dirty, so I'd love to get input on what other people did :yum: 

(My solution is also very slow.. #38)

**Spoilers below!**

## The mistake I made

When looking for locations that could count towards the area of locations that is closest to a finite coordinate I search in a square from the left-most x and y of any coordinate to the right-most x and y of any seen coordinate. So given three coordinates A, B and C (and some excluded ones below that make C finite) the search area could look like this:

```
0: not searched
x: search area

0 0 0 0 0 0 0 0 0
0 A x x x x x B 0
0 x x x x x x x 0
0 x x x C x x x 0
...(more coordinates below)
```

My solution would see A and B, and determine that the only area it needs to search for locations closest to C is the area marked as x. 0 means the location is never looked at. This works out fine in this example, but here is an example that does not work with my solution:

```
0 0 0 0 0 0 ! 0 0 0 0 0 0
0 A x x x x x x x x x C 0
0 x x x x x C x x x x x 0
...(more coordinates below)
```

In this example the location marked as "!" is closest to C, so it should be counted towards it's area for the solution of part 1. My code only looks at the area marked as x though, so it would get the answer wrong here.

## A fix

A solution would be for the code to search through the area in a diamond shape like this:

```
0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 x 0 0 0 0 0 0
0 0 0 0 0 x x x 0 0 0 0 0
0 0 0 0 x x x x x 0 0 0 0
0 0 0 x x x x x x x 0 0 0
0 0 x x x x x x x x 0 0 0
0 A x x x x x x x x x C 0
0 x x x x x C x x x x x 0
...(more coordinates below)
```

This would catch the example above, and (I think) it would catch any other such edge cases. This fix seems a little dirty though, and would require over double the area to be searched...

Let me know if anything is unclear. Any input is awesome. Guessing someone found a smarter way of searching :)

**Comment by Arxcis** [comment]
I don't think your diamond-solution is enough. Your area C is also infinite in the situation you describe 
above.

Consider:
```
- - - - - - - - - - - - - - -
- - - - - - - - - - - - - - -
- - A a a a . c . b b b B - -
- - a a a . c c c . b b b - -
- - a a . c c c c c . b b - -
- - a . c c c C c c c . b - -
- - a . c c c c c c c . b - -
- - . . . . . . . . . . b - -
- - . d d d d d d d d d . - -
- - d d d d d D d d d d d - -
- - - - - - - - - - - - - - -
- - - - - - - - - - - - - - -
```

As long as any point of the area C is an extreme point on the perimeter, C will be an infinite.

```
- - a a a a . c . b b b b - -
- - a a a a . c . b b b b - -
- - a a a a . c . b b b b - -
- - a a a a . c . b b b b - -
- - a a a a . c . b b b b - -
- - a a a a . c . b b b b - -
- - a a a a . c . b b b b - -
- - a a a a . c . b b b b - -
- - a a a a . c . b b b b - -
- - a a a a . c . b b b b - -
- - A a a a . c . b b b B - -
- - a a a . c c c . b b b - -
- - a a . c c c c c . b b - -
- - a . c c c C c c c . b - -
- - a . c c c c c c c . b - -
- - . . . . . . . . . . b - -
- - . d d d d d d d d d . - -
- - d d d d d D d d d d d - -
- - - - - - - - - - - - - - -
- - - - - - - - - - - - - - -

```

**Comment by tholok97** [comment]
Yea! That seems right. Looking into your cpp solution now to see how you solved it.

**Comment by tholok97** [comment]
This part blew my mind:
```
    const auto points = ([](){
        auto _points = std::vector<Point>{};
        for (std::string line; std::getline(std::cin, line); ) {
            auto p = Point{};
            sscanf(line.c_str(), "%d, %d", &p.x, &p.y);
            _points.emplace_back(p);
        }
        return _points;
    })();
```

:yum: 

But yea I understand yours now. Thanks! I'll try and implement something similar.

You use a two-dimentional array of structs though, and Bash has neither two-dimentional arrays nor structs, so will have to trick it into doing what I want using other means :laughing:

**Comment by tholok97** [comment]
Solved by #44
